src/components/ListingDetails/index.tsx

```diff
@@ -0,0 +1,399 @@
+/* eslint-disable react/jsx-no-comment-textnodes */
+import React, { useMemo } from 'react';
+import {
+  formatTimestamp,
+  parseAgency,
+  parseTimeCommitment,
+} from '@/utils/helpers';
+import { H2, H3, H4, P, StrongP } from '@/styles/text';
+import {
+  CaseListing,
+  DocumentTranslation,
+  Interpretation,
+  LimitedCaseAssignment,
+  Listing,
+} from '@/types/schema';
+import { useProfile } from '@/utils/ProfileProvider';
+import { useAuth } from '@/utils/AuthProvider';
+import COLORS from '@/styles/colors';
+import InterestForm from '../InterestForm';
+import { LinkButton } from '../Button';
+import Icon from '../Icon';
+import {
+  CaseDisplay,
+  InfoContainer,
+  FieldContainer,
+  InnerFieldContainer,
+  AuthButtons,
+  IconTextGroup,
+  BorderedSection,
+  Subheading,
+} from './styles';
+
+const renderField = (label: string, value: string | boolean | number) => (
+  <FieldContainer>
+    <StrongP>{label}</StrongP>
+    <P>{value}</P>
+  </FieldContainer>
+);
+
+interface ListingField<T extends Listing> {
+  label: string;
+  getValue: (data: T) => string;
+}
+
+const caseFields: ListingField<CaseListing>[] = [
+  // Relief sought
+  {
+    label: 'Relief Sought',
+    getValue: data => data.relief_codes.join(', ') || 'N/A',
+  },
+  // Time Commitment
+  {
+    label: 'Time Commitment',
+    getValue: data => parseTimeCommitment(data.hours_per_week, data.num_weeks),
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: data => (data.is_remote ? 'Remote' : 'In Person'),
+  },
+  // Adjudicating Agency
+  {
+    label: 'Adjudicating Agency',
+    getValue: data =>
+      data.adjudicating_agency ? parseAgency(data.adjudicating_agency) : 'N/A',
+  },
+  // Client Languages
+  {
+    label: 'Client Language(s)',
+    getValue: data => data.languages.join(', '),
+  },
+  // Client Country of Origin
+  {
+    label: 'Client Country of Origin',
+    getValue: data => data.country || 'N/A',
+  },
+  // Client vs. Custody Location handled in Component
+];
+
+const caseInterpretationFields: ListingField<CaseListing>[] = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: data => data.languages.join(', '),
+  },
+  // Time Commitment
+  {
+    label: 'Time Commitment',
+    getValue: data => parseTimeCommitment(data.hours_per_week, data.num_weeks),
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: data => (data.is_remote ? 'Remote' : 'In Person'),
+  },
+];
+
+const intepretationFields: ListingField<Interpretation>[] = [
```

Comment:
nit: should this be `interpretationFields`? (minor typo)

Reply:
yes! good catch